hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -217,18 +218,52 @@ public static HoodieTableMetaClient initTableIfNotExists(
           .setCDCEnabled(conf.getBoolean(FlinkOptions.CDC_ENABLED))
           .setCDCSupplementalLoggingMode(conf.getString(FlinkOptions.SUPPLEMENTAL_LOGGING_MODE))
           .setTimelineLayoutVersion(1)
+          .setHoodieIndexConf(OptionsResolver.getHoodieIndexConf(conf))
           .initTable(hadoopConf, basePath);
       LOG.info("Table initialized under base path {}", basePath);
       return metaClient;
     } else {
       LOG.info("Table [{}/{}] already exists, no need to initialize the table",
           basePath, conf.getString(FlinkOptions.TABLE_NAME));
-      return StreamerUtil.createMetaClient(basePath, hadoopConf);
+      HoodieTableMetaClient client = StreamerUtil.createMetaClient(basePath, hadoopConf);
+      validateTableConfig(conf, client.getTableConfig());
```

Comment:
Did you notice that the tabe can only be initialized only once?

Reply:
We just need to verify when the StreamWriteOperatorCoordinator#start or HoodieTableFactory#setupTableOptions. Just ensure that the job cannot be started if the validation fails.

Reply:
Yeah, I notice that `the bucket num` is not even a table config, so how could we know the preceding num buckets then ?

Reply:
I have already stored `HoodieIndexConfig` in the table configuration when initializing the table for the first time.

Reply:
ok, I think it can be promoted from my perspective.

Reply:
from my perspective, i think it can be promoted.

Reply:
We need to fire another issue first, to discuss promote the bucket number as table config instead of a write config, after that, we can continue with this patch.